src/battle.ts

```diff
@@ -1409,33 +1409,42 @@ export class Battle {
 		this.scene.updateStatbar(pokemon);
 		if (fromeffect.id === 'sleeptalk') {
 			pokemon.rememberMove(move.name, 0);
-		} else if (!fromeffect.id || fromeffect.id === 'pursuit') {
+		}
+		let callerMoveForPressure = null;
+		// will not include effects that are conditions named after moves like Magic Coat and Snatch, which is good
+		if (fromeffect.id && kwArgs.from.startsWith("move:")) {
+			callerMoveForPressure = fromeffect as Move;
+		}
+		if (!fromeffect.id || callerMoveForPressure || fromeffect.id === 'pursuit') {
 			let moveName = move.name;
-			if (move.isZ) {
-				pokemon.item = move.isZ;
-				let item = Dex.items.get(move.isZ);
-				if (item.zMoveFrom) moveName = item.zMoveFrom;
-			} else if (move.name.slice(0, 2) === 'Z-') {
-				moveName = moveName.slice(2);
-				move = Dex.moves.get(moveName);
-				if (window.BattleItems) {
-					for (let item in BattleItems) {
-						if (BattleItems[item].zMoveType === move.type) pokemon.item = item;
+			if (!callerMoveForPressure) {
+				if (move.isZ) {
+					pokemon.item = move.isZ;
+					let item = Dex.items.get(move.isZ);
+					if (item.zMoveFrom) moveName = item.zMoveFrom;
+				} else if (move.name.slice(0, 2) === 'Z-') {
+					moveName = moveName.slice(2);
+					move = Dex.moves.get(moveName);
+					if (window.BattleItems) {
+						for (let item in BattleItems) {
+							if (BattleItems[item].zMoveType === move.type) pokemon.item = item;
+						}
 					}
 				}
 			}
 			let pp = 1;
 			// Sticky Web is never affected by pressure
 			if (this.abilityActive(['Pressure']) && move.id !== 'stickyweb') {
```

Comment:
```suggestion
			if (this.abilityActive(['Pressure'])) {
```
I would remove this hardcode since it would get handled by `move.pressureTarget`.